### ISSUE TITLE
GPU resource and how to create data

### ISSUE THREAD

**Issue by clearwaterkzk** [mikelunizar/Plasticity-MeshGraphNets#2]
Hi, thanks for sharing nice code.
I also read your meshGraphNets_pytorch code for deforming plate. 
https://github.com/mikelunizar/meshGraphNets_pytorch

I have three questions.
- What GPU did you use and what time did it take to train till validation loss converge for your  and deepmind dataset ?
- How did you make plots/collision,gif ? 
- What software did you create train/val/test data ? Open Souce simulator like OpenFOAM, OpenRadioss ?

Thank you.

**Comment by mikelunizar** [comment]
Hi clearwaterkzk,

Thanks for your interest!

The forked code gave me quite a few issues when trying to run it, 
especially for the deforming plate case. So I didn’t actually use it for 
training. Instead, I mainly used it to download and inspect DeepMind's 
dataset to understand its structure.

For the GPU, I used an RTX 4090 with 24GB of memory.

The plots were rendered using the PyVista library.

As for the dataset, it was generated using our in-house FEM code 
developed in our lab. However, it doesn’t differ much from commercial 
software like Abaqus.

I hope this clarifies things! I'll soon release a new, more 
user-friendly and efficient code on my GitHub, for wave propagation, 
diffusion, and collision simulations. Feel free to check it out once 
it's up!

Best regards,
Mikel

**Comment by clearwaterkzk** [comment]
Thanks for detail !!

> The forked code gave me quite a few issues when trying to run it,
> especially for the deforming plate case. So I didn’t actually use it for
> training. Instead, I mainly used it to download and inspect DeepMind's
> dataset to understand its structure.
What problem did you have ?
I'm reading some codes including yours and trying to reproduce the result about deforming plate, but 

The design of base block or model in your forked code (meshGraphNets_pytorch) doesn't seem to be significantly different from your Plasticity-MeshGraphNets.
Did you reproduce deepmind's performance about deforming plate ?

**Comment by clearwaterkzk** [comment]
Thanks for detail !!

> The forked code gave me quite a few issues when trying to run it,
> especially for the deforming plate case. So I didn’t actually use it for
> training. Instead, I mainly used it to download and inspect DeepMind's
> dataset to understand its structure.

What problem did you have ?
I'm reading some codes including yours and trying to reproduce the result about deforming plate, but I'm encountering some issues.
The training code like your train_dp.py gives me similar one-step result to deepmind's but the RMSE during evaluation(rollout) is quite poor.
This might be due to undertraining, so I think it’s necessary to monitor the validation loss during the training loop like your pytorch lightning training.

The design of base block or model in your forked code (meshGraphNets_pytorch) doesn't seem to be significantly different from your Plasticity-MeshGraphNets.
Did you reproduce deepmind's performance about deforming plate ?

Thank you for your time.

**Comment by mikelunizar** [comment]
I recommend you to use and test the code available in the Plasticity-MeshGraphNets, the code is curated and works for collisions. We researched the vanilla training process of the plate-actuator collision and also other scenarios, as transfer learning with MeshGraphNets and collisions in plasticity, quite extrapolable to other materials.
 
From my experience you can achieve pretty good results with the appropriated hyper parameter and training time, and as you mentioned, it's crucial to monitor the losses in order to track the performance of the model. 

As mentioned, new code will be available soon! All the luck!
